recipes/vorbis/all/patches/link-libm-find-package-ogg.patch

```diff
@@ -0,0 +1,56 @@
+Backport in 1.3.6 some build improvements of 1.3.7:
+* link to libm
+* usage of find_package to discover Ogg and link with Ogg:ogg target
+
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -4,6 +4,7 @@ project(vorbis)
+ # Required modules
+ include(GNUInstallDirs)
+ include(CheckIncludeFiles)
++include(CheckLibraryExists)
+ 
+ # Build options
+ option(BUILD_SHARED_LIBS "Build shared library" OFF)
+@@ -52,18 +53,12 @@ endfunction()
+ 
+ message(STATUS "Configuring ${PROJECT_NAME} ${PROJECT_VERSION}")
+ 
++# Find math library
++
++check_library_exists(m floor "" HAVE_LIBM)
++
+ # Find ogg dependency
+-if(NOT OGG_ROOT)
```

Comment:
may this large patch be avoided? (e.g. by using pkg-config generator?)

Reply:
It would be more complex, since the purpose of this patch is to have the same behavior than 1.3.7

Reply:
do we have an issue for pkg-config shared?

Reply:
It's not specific to pkgconfig, but all recipes used as build requirements and potentially depending on a shared lib. It's a known issue due to SIP, I don't know if there is a specific issue in conan side.